enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NullListAcceptanceTest.scala

```diff
@@ -0,0 +1,159 @@
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.internal.cypher.acceptance
+
+import org.neo4j.cypher._
+import org.neo4j.internal.cypher.acceptance.CypherComparisonSupport.Versions.{V2_3, V3_2}
+import org.neo4j.internal.cypher.acceptance.CypherComparisonSupport._
+
+class NullListAcceptanceTest extends ExecutionEngineFunSuite with CypherComparisonSupport {
+
+  // Changed behaviour to comply with opencypher in 3.3
+  val nullInListConfigOld: TestConfiguration = TestConfiguration(V2_3 -> V3_2, Planners.all, Runtimes.all) + Configs.AllRulePlanners
+
+  // Comparison between lists and non-lists
+
+  test("equality between list and literal should return false") {
+    val query = "WITH [1, 2] AS l1, 'foo' AS l2 RETURN l1 = l2 AS res"
+
+    val result = executeWith(Configs.All, query)
+
+    result.toList should equal(List(Map("res" -> false)))
+  }
+
+  // Equality between lists with null
+
+  test("equality of lists of different length should return false despite nulls") {
+    val query = "WITH [1] AS l1, [1, null] AS l2 RETURN l1 = l2 AS res"
+
+    val result = executeWith(Configs.All, query)
+
+    result.toList should equal(List(Map("res" -> false)))
+  }
+
+  test("equality between different lists with null should return false") {
+    val query = "WITH [1, 2] AS l1, [null, 'foo'] AS l2 RETURN l1 = l2 AS res"
+
+    val result = executeWith(Configs.All, query)
+
+    result.toList should equal(List(Map("res" -> false)))
+  }
+
+  test("equality between almost equal lists with null should return null") {
+    val query = "WITH [1, 2] AS l1, [null, 2] AS l2 RETURN l1 = l2 AS res"
+
+    val result = executeWith(Configs.All, query, expectedDifferentResults = nullInListConfigOld)
+
+    result.toList should equal(List(Map("res" -> null)))
+  }
+
+  // Nested Lists
+  test("equality of nested lists of different length should return false despite nulls") {
+    val query = "WITH [[1]] AS l1, [[1], [null]] AS l2 RETURN l1 = l2 AS res"
+
+    val result = executeWith(Configs.All, query)
+
+    result.toList should equal(List(Map("res" -> false)))
+  }
+
+  test("equality between different nested lists with null should return false") {
+    val query = "WITH [[1, 2], [1, 3]] AS l1, [[1, 2], [null, 'foo']] AS l2 RETURN l1 = l2 AS res"
+
+    val result = executeWith(Configs.All, query)
+
+    result.toList should equal(List(Map("res" -> false)))
+  }
+
+  test("equality between almost equal nested lists with null should return null") {
+    val query = "WITH [[1, 2], ['foo', 'bar']] AS l1, [[1, 2], [null, 'bar']] AS l2 RETURN l1 = l2 AS res"
+
+    val result = executeWith(Configs.All, query, expectedDifferentResults = nullInListConfigOld)
+
+    result.toList should equal(List(Map("res" -> null)))
+  }
+
+  // IN with null
+
+  test("IN with different length lists should return false despite nulls") {
+    val query = "WITH [1] AS l1, [1, null] AS l2 RETURN l1 IN [l2] AS res"
```

Comment:
Can we change that to `WITH [1] AS l1, [[1, null]] AS l2 RETURN l1 IN l2 AS res` to make it consistent with the other tests?